apps/internal/oauth/ops/authority/authority.go

```diff
@@ -23,7 +23,8 @@ import (
 
 const (
 	authorizationEndpoint             = "https://%v/%v/oauth2/v2.0/authorize"
-	instanceDiscoveryEndpoint         = "https://%v/common/discovery/instance"
+	aadInstanceDiscoveryEndpoint      = "https://%v/common/discovery/instance"
+	dstsInstanceDiscoveryEndpoint     = "https://%v/dstsv2/common/discovery/instance"
```

Comment:
>https://%v/dstsv2/common/discovery/instance [](http://example.com/codeflow?start=38&length=43)

It seems consuming this endpoint requires changes on the dSTS side. I believe this is the first time a client takes dependency on this endpoint with dSTS. 

If you see the output of the endpoint using a test dSTS instance you will notice that the metadata returned is incorrect and it points mostly to AAD.

Connect to the VPN before loading this on your browser:

[[https://co2agg04-dsts.dsts.core.azure-test.net/dstsv2/common/discovery/instance?api-version=1.1&authorization_endpoint=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize](https://co2agg04-dsts.dsts.core.azure-test.net/dstsv2/common/discovery/instance?api-version=1.1&authorization_endpoint=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize)](https://co2agg04-dsts.dsts.core.azure-test.net/dstsv2/common/discovery/instance?api-version=1.1&authorization_endpoint=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize)

This is the response

```json
{
    "tenant_discovery_endpoint": "[https://login.microsoftonline.com/dstsv2/common/v2.0/.well-known/openid-configuration](https://login.microsoftonline.com/dstsv2/common/v2.0/.well-known/openid-configuration)",
    "api-version": "1.1",
    "metadata": [
        {
            "preferred_network": "login.microsoftonline.com",
            "preferred_cache": "login.windows.net",
            "aliases": [
                "login.microsoftonline.com",
                "login.windows.net",
                "login.microsoft.com",
                "sts.windows.net"
            ]
        },
        {
            "preferred_network": "login.partner.microsoftonline.cn",
            "preferred_cache": "login.partner.microsoftonline.cn",
            "aliases": [
                "login.partner.microsoftonline.cn",
                "login.chinacloudapi.cn"
            ]
        },
        {
            "preferred_network": "login.microsoftonline.de",
            "preferred_cache": "login.microsoftonline.de",
            "aliases": [
                "login.microsoftonline.de"
            ]
        },
        {
            "preferred_network": "login.microsoftonline.us",
            "preferred_cache": "login.microsoftonline.us",
            "aliases": [
                "login.microsoftonline.us",
                "login.usgovcloudapi.net"
            ]
        },
        {
            "preferred_network": "login-us.microsoftonline.com",
            "preferred_cache": "login-us.microsoftonline.com",
            "aliases": [
                "login-us.microsoftonline.com"
            ]
        },
        {
            "preferred_network": "co2agg04-passive-dsts.dsts.core.azure-test.net/dstsv2",
            "preferred_cache": "co2agg04-passive-dsts.dsts.core.azure-test.net/dstsv2",
            "aliases": [
                "co2agg04-passive-dsts.dsts.core.azure-test.net"
            ]
        }
    ]
}
```

Reply:
removed